File: pset1/mario/less/mario.c
#include <cs50.h>
#include <stdio.h>
#include <cs50.h>

int main(void)

{
    int rows;
    int spaces;
    int hashes;
    //get number of rows required from user
    do
    {
        rows = get_int("Please enter a number of blocks between 1 and 23: ");
    }
    //continue with loop to get number of rows if a valid number isn't entered
    while (rows < 0 || rows > 23);

    //loop for the spaces and dashes per row
    for (int i = 1; i <= rows; i++)
    {
        for (spaces = (rows - i); spaces > 0; spaces--)
        {
            printf(" ");
        }
        for (hashes = 1; hashes <= (i + 1); hashes++)
        {
            printf("#");
        }

        printf("\n");
    }
    return 0;
}

File: pset6/mario/less/mario.py
import cs50

# get integer from user and check for value between 0 and 23
while True:
    rows = cs50.get_int("Please enter a number of blocks between 1 and 23: ")
    if rows >= 0 and rows <= 23:
        break
# create half pyramid based in number of rows input by user
for i in range(1, (rows + 1)):
    print(" " * (rows - i) + "#" * (i + 1))